components/utils/isvisable.ts

```diff
@@ -1,21 +1,22 @@
-import { useEffect, useState } from 'react'
+import { useEffect, useState, RefObject } from 'react'
 
-const OPTIONS = {
+const OPTIONS: IntersectionObserverInit = {
   root: null,
   rootMargin: '0px 0px 0px 0px',
   threshold: 0,
 }
 
-const useIsVisable = (elementRef) => {
+const useIsVisable = (elementRef: RefObject<HTMLElement>) => {
   const [isVisable, setIsVisable] = useState(false)
 
   useEffect(() => {
     if (elementRef.current) {
+      const currentElement = elementRef.current
       const observer = new IntersectionObserver((entries, observer) => {
         entries.forEach((entry) => {
           if (entry.isIntersecting) {
             setIsVisable(true)
-            observer.unobserve(elementRef.current)
+            observer.unobserve(currentElement)
```

Comment:
introduced `currentElement` as it will be proved to not be `undefined` by this point.